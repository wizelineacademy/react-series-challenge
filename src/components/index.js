import Card from './Card'
import Detail from './Detail'
import GifItem from './GifItem'
import Grid from './Grid'
import Header from './Header'
import Home from './Home'
import SearchBar from './SearchBar'

export {
  Card,
  Detail,
  GifItem,
  Grid,
  Header,
  Home,
  SearchBar
}
