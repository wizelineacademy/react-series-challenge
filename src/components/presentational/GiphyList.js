import React from 'react';
import {NavLink} from 'react-router-dom';
import styled from 'styled-components';

const MainContainer = styled.div`
  width: 100%;
`;

const GiphyListContainer = styled.ul`
  list-style: none;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  -webkit-flex-flow: row wrap;
  justify-content: space-around;

  @media (max-width: 800px) {
    justify-content: space-around;
  }

  @media (max-width: 500px) {
    flex-direction: column;
  }
`;

export const GiphyListItem = styled.li`
  display: list-item;
  position: relative;
  text-align: -webkit-match-parent;
  width: 29%;
  height: 250px;
  border: 1.5px solid lightgrey;
  margin-right: 20px;
  margin-bottom: 20px;
  border-radius: 10px;
  padding: 8px;
  
  @media (max-width: 1024px) {
    width: 42%;
  }

  @media (max-width: 615px) {
    width: 90%;
  }
`;

export const GiphyListFullItem = styled.li`
  display: list-item;
  text-align: -webkit-match-parent;
  width: 100%;
  height: 290px;
  font-weight: bold;
  font-size: 1.25em;
  display: flex;
  justify-content: center;
  align-items: center;
`;

export const GiphyListItemHeader = styled.header`
  
  width: 100%;
  height: 15%;
  font-size: 1.2em;
  display: flex;
  justify-content: center;
  align-items: center;
  text-align: center;
  border-bottom: 1px solid lightgray;
  @media (max-width: 600px) {
    
  }
`;

export const GiphyListItemBody = styled.section`
  
  width: 100%;
  height: 75%;
  font-size: 1.2em;
  display: flex;
  justify-content: center;
  align-items: center;
  text-align: center;
  @media (max-width: 600px) {
    
  }

  img {
    width: 100%;
    height: 100%
  }
`;

export const GiphyListItemFooter = styled.footer`
  
  width: 100%;
  height: 10%;
  font-size: 1.2em;
  display: flex;
  justify-content: flex-end;
  align-items: flex-end;
  text-align: right;
  border-top: 1px solid lightgray;
  @media (max-width: 600px) {
    
  }
`;

export const GiphyLinkFavorite = styled.a`
  color: inherit;
  text-decoration: inherit;
`;

export const GiphyHeartFavorite = styled.span`
  background-color: ${props => props.color ? props.color : 'lightgray'};
  display: inline-block;
  height: 15px;
  margin: 0 10px;
  position: relative;
  top: 3px;
  transform: rotate(-45deg);
  width: 15px;


  &:before, &:after {
    content: '';
    background-color: ${props => props.color ? props.color : 'lightgray'};
    border-radius: 50%;
    height: 15px;
    position: absolute;
    width: 15px;
  }

  &:before {
    top: -7.5px;
    left: 0;
  }

  &:after {
    left: 7.5px;
    top: 0;
  }
`;



const GiphyListPager = styled.ul`
  position:relative;
  list-style: none;
  display: ${props => props.visible ? '-webkit-box' : 'none'};
  display: ${props => props.visible ? '-moz-box' : 'none'};
  display: ${props => props.visible ? '-ms-flexbox' : 'none'};
  display: ${props => props.visible ? '-webkit-flex' : 'none'};
  -webkit-flex-flow: row wrap;
  display: ${props => props.visible ? 'flex' : 'none'};
  flex-flow: row wrap;
  justify-content: center;
  background: #ff9800;
  height: 50px;
  width:101%;
  margin-left: -15px;

  @media (max-width: 800px) {
    justify-content: space-around;
  }

  @media (max-width: 500px) {
    flex-direction: column;
  }
`;

const GiphyListPagerItem = styled.li`
  display: list-item;
  text-align: -webkit-match-parent;
  height: 50px;
  
  @media (max-width: 1024px) {
    width: 46%;
  }

  @media (max-width: 600px) {
    width: 97%;
  }
  & div {
    margin-top: 20px;
    font-size: 1.25em;
  }
`;

const Button = styled.button`
  font-size: 1em;
  margin: .6em;
  padding: 0.25em 1em;
  border-radius: 3px;
`;


const Link = styled(NavLink)`
  width: 100%;
  height: 100%;
  text-decoration: none;
  color: inherit;
  &:hover {
    outline-width: 0;
  }

  &:active {
    outline-width: 0;
  }
`;

const Item = ({source, clickAdd2Favorites = () =>  { }}) => {
  /*eslint no-script-url:0*/
  return (
    <GiphyListItem source={source}>
      <GiphyListItemHeader>
        <h3>{source.title}</h3>
      </GiphyListItemHeader>
      <GiphyListItemBody>
        <Link to={`/gifs/${source.id}`}>
          <img src={source.images.fixed_width.url} alt={source.title} />
        </Link>
      </GiphyListItemBody>
      <GiphyListItemFooter>
        <GiphyLinkFavorite href='javascript:void(0)' onClick={(el) => {
          clickAdd2Favorites(source, el.target);
        }} >
          {source.isFavorite === true
              ? <span>Remove from my favorites</span> 
              : <span>Send to my favorites</span>}
          <GiphyHeartFavorite color={source.isFavorite ? 'tomato' : 'lightgray'} />
        </GiphyLinkFavorite>
      </GiphyListItemFooter>
    </GiphyListItem>
  );
};

export const GiphyListPagerButton = ({label, onClick}) => {
  return (
    <GiphyListPagerItem>
      <Button onClick={onClick}>{label}</Button>
    </GiphyListPagerItem>
  );
};

const GiphyList = (props) => {
  const { dataSource, clickAdd2Favorites } = props;

  const renderItems = ({data, fetching}) => {
    if (!fetching) {
      if (data.length > 0) {
        return data.map((giphyItem) => {
          return <Item 
            source={giphyItem}
            key={giphyItem.id}
            clickAdd2Favorites={clickAdd2Favorites} />;
        });
      } else {
        return <GiphyListFullItem>Not found gifs ...</GiphyListFullItem>;
        
      }
    } else {
      return <GiphyListFullItem>Loading gifs ...</GiphyListFullItem>;
    }
  };
  
  return (
    <MainContainer>
      <GiphyListContainer>
        {renderItems(dataSource)}
      </GiphyListContainer>
      <GiphyListPager visible={!dataSource.fetching && dataSource.data.length > 0}>
        <GiphyListPagerButton label='First' onClick={props.onClickFirstPage} />
        <GiphyListPagerButton label='Previous' onClick={props.onClickPreviousPage} />
        <GiphyListPagerItem>
          <div>
              Current page: 
              <strong> {dataSource.pagination.offset} / {dataSource.pagination.pages}</strong>
          </div>
        </GiphyListPagerItem>
        <GiphyListPagerButton label='Next' onClick={props.onClickNextPage} />
        <GiphyListPagerButton label='Last' onClick={props.onClickLastPage} />
      </GiphyListPager>
    </MainContainer>
  );
};

export default GiphyList;
