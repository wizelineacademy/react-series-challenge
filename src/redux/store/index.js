import { createStore, applyMiddleware, compose } from 'redux';
import rootReducer from '../reducers';
import customMiddleware from '../middleware';
import createSagaMiddleware from 'redux-saga';
import rootSaga from '../sagas';

const sagaMiddleware = createSagaMiddleware();

const initialState = {};
const enhancers = [];
const middleware = [
  sagaMiddleware,
  ...customMiddleware,
];

const composedEnhancers = compose(
  applyMiddleware(...middleware),
  ...enhancers,
);

const store = createStore(
  rootReducer,
  initialState,
  composedEnhancers,
);

sagaMiddleware.run(rootSaga);

export default store;
