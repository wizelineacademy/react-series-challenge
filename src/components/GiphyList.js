import React from 'react';
import {bindActionCreators} from 'redux';
import {connect} from 'react-redux';
import GiphyElement from './GiphyElement';
import ErrorBoundary from './ErrorBoundary';
import appActions from '../actions';

const GiphyList = (props) => {

    const {
        view,
        giphies,
        favorites,
        addFavorites,
        removeFavorites
    } = props;

    const giphiesShown = (view === appActions.enums.viewType.SEARCH)
        ? giphies
        : favorites;

    return (
        <React.Fragment>
            {giphiesShown.map(giphy =>
                <ErrorBoundary key={giphy.id}>
                    <GiphyElement
                        key={giphy.slug}
                        giphy={giphy}
                        view={view}
                        addFavorites={addFavorites}
                        removeFavorites={removeFavorites}
                    />
                </ErrorBoundary>)}
        </React.Fragment>
    );

};

const mapStateToProps = (state) => {
    const {
        view,
        giphies,
        favorites
    } = state;
    return {
        view,
        giphies,
        favorites
    };
};

const mapDispatchToProps = (dispatch) => {
    const {
        addFavorites,
        removeFavorites
    } = appActions.creators;
    return bindActionCreators({
        addFavorites,
        removeFavorites
    }, dispatch);
};

export default connect(mapStateToProps, mapDispatchToProps)(GiphyList);
