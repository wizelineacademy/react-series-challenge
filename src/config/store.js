import {createStore, applyMiddleware, compose} from 'redux';
import createSagaMiddleware from 'redux-saga';

import rootReducer from '../reducers';
import rootSaga from '../sagas';
import {APPLICATION_LOCAL_STORAGE_KEY} from './config';
import {initialState as gifInitialState} from '../reducers/gif';

// Middlewares
const sagaMiddleware = createSagaMiddleware();
const middlewares = [
  sagaMiddleware
];

// Enhancers
const createEnhancers = () => {
  let enhancers = [];

  /**
   * Uncomment this lines and add the redux-devtools package to enable redux dev
   *  tools into development mode.
   */
  // if (process.env.NODE_ENV !== 'production') {
  //   enhancers.push(
  //     window.__REDUX_DEVTOOLS_EXTENSION__
  //       && window.__REDUX_DEVTOOLS_EXTENSION__());
  // }

  return enhancers;
};

const persistStateDetails = (state) => {
  try {
    let stateCopy = JSON.parse(JSON.stringify(state));

    // Remove unnecesary state details
    stateCopy.gifs.fetching = gifInitialState.fetching;
    stateCopy.gifs.displayed = gifInitialState.displayed;
    stateCopy.gifs.favoritesTextFilter = gifInitialState.favoritesTextFilter;

    let serializedState = JSON.stringify(stateCopy);
    localStorage.setItem(APPLICATION_LOCAL_STORAGE_KEY, serializedState);
  } catch (e) {}
}

const retrievePreviousStateDetails = () => {
  let result = {};

    try {
      let serializedState =
        window.localStorage.getItem(APPLICATION_LOCAL_STORAGE_KEY);

      if (serializedState) {
        result = JSON.parse(serializedState);
      }
    } catch (e) {}

  return result;
}

const enhancers = createEnhancers();
const composedEnhancers = (enhancers.length > 0)
  ? compose(applyMiddleware(...middlewares), ...enhancers)
  : applyMiddleware(...middlewares);


const initializeStore = (initialState) => {
  if (!initialState) {
    // Rehydrate the store
    initialState = retrievePreviousStateDetails();
  }

  // Store object
  const store = createStore(rootReducer, initialState, composedEnhancers);

  store.subscribe(() => persistStateDetails(store.getState()));

  sagaMiddleware.run(rootSaga);

  return store;
}

export default initializeStore;
