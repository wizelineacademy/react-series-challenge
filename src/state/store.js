import createSagaMiddleware from "redux-saga";
import { createStore, applyMiddleware } from "redux";

import rootSaga from "./sagas";
import reducers from "./reducers";

const sagaMiddleware = createSagaMiddleware();

const store = createStore(reducers, applyMiddleware(sagaMiddleware));

sagaMiddleware.run(rootSaga);

export default store;
