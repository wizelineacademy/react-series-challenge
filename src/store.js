import { createStore, applyMiddleware, combineReducers } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';

import { gif, trending } from './reducers';

const reducer = combineReducers({ gif, trending });

export default createStore(reducer, composeWithDevTools(applyMiddleware(thunk)));
