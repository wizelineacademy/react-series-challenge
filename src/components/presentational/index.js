import Menu from './Menu';
import ItemMenu from './ItemMenu';
import GiphyList from './GiphyList';
import InputSearch from './InputSearch';
import ErrorWatcher from './ErrorWatcher';

export {
  Menu,
  ItemMenu,
  GiphyList,
  InputSearch,
  ErrorWatcher
};