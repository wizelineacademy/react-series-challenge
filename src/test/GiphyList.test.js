import React from 'react';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { Provider } from 'react-redux';
import {MemoryRouter as Router} from 'react-router-dom';
import store from '../store';
import {GiphyList} from '../components/presentational';
import {GiphyLinkFavorite, GiphyHeartFavorite} from '../components/presentational/GiphyList';
import dataSource from './__mocks/completeFavoritesDataSource';

describe('GiphyList', () => {
  it('should render correctly', () => {
    const output = shallow(
      <Provider store={store}>
        <Router>
          <GiphyList dataSource={dataSource} />
        </Router>
      </Provider>
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('Controls structure', () => {
    const wrapper = mount(
      <Provider store={store}>
        <Router>
          <GiphyList dataSource={dataSource} />
        </Router>
      </Provider>
    );
    const ul = wrapper.find('ul');
    
    expect(ul.exists()).toEqual(true);
    
    expect(wrapper.find(GiphyLinkFavorite)).toHaveLength(2);
    const link = wrapper.find(GiphyLinkFavorite).at(0);
    expect(link.exists()).toEqual(true);
    link.simulate('click');

  });

  it('GiphyHeartFavorite', () => {
    const wrapper = mount(
      <GiphyHeartFavorite color='green' />
    );
    const span = wrapper.find('span');
    
    expect(span.exists()).toEqual(true);
  });
});
