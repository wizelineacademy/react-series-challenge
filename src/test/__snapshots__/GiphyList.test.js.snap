// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GiphyList should render correctly 1`] = `
<MemoryRouter>
  <GiphyList
    dataSource={
      Object {
        "data": Array [
          Object {
            "bitly_gif_url": "https://gph.is/2qcEv2M",
            "bitly_url": "https://gph.is/2qcEv2M",
            "content_url": "",
            "embed_url": "https://giphy.com/embed/3bztNGb4XTnwxOgotG",
            "id": "3bztNGb4XTnwxOgotG",
            "images": Object {
              "480w_still": Object {
                "height": "270",
                "url": "https://media0.giphy.com/media/3bztNGb4XTnwxOgotG/480w_s.jpg",
                "width": "480",
              },
              "downsized": Object {
                "height": "270",
                "size": "1556485",
                "url": "https://media0.giphy.com/media/3bztNGb4XTnwxOgotG/giphy.gif",
                "width": "480",
              },
              "downsized_large": Object {
                "height": "270",
                "size": "1556485",
                "url": "https://media0.giphy.com/media/3bztNGb4XTnwxOgotG/giphy.gif",
                "width": "480",
              },
              "downsized_medium": Object {
                "height": "270",
                "size": "1556485",
                "url": "https://media0.giphy.com/media/3bztNGb4XTnwxOgotG/giphy.gif",
                "width": "480",
              },
              "downsized_small": Object {
                "height": "270",
                "mp4": "https://media0.giphy.com/media/3bztNGb4XTnwxOgotG/giphy-downsized-small.mp4",
                "mp4_size": "191415",
                "width": "480",
              },
              "downsized_still": Object {
                "height": "270",
                "size": "95193",
                "url": "https://media0.giphy.com/media/3bztNGb4XTnwxOgotG/giphy-downsized_s.gif",
                "width": "480",
              },
              "fixed_height": Object {
                "height": "200",
                "mp4": "https://media0.giphy.com/media/3bztNGb4XTnwxOgotG/200.mp4",
                "mp4_size": "75176",
                "size": "839125",
                "url": "https://media0.giphy.com/media/3bztNGb4XTnwxOgotG/200.gif",
                "webp": "https://media0.giphy.com/media/3bztNGb4XTnwxOgotG/200.webp",
                "webp_size": "172568",
                "width": "356",
              },
              "fixed_height_downsampled": Object {
                "height": "200",
                "size": "316095",
                "url": "https://media0.giphy.com/media/3bztNGb4XTnwxOgotG/200_d.gif",
                "webp": "https://media0.giphy.com/media/3bztNGb4XTnwxOgotG/200_d.webp",
                "webp_size": "61638",
                "width": "356",
              },
              "fixed_height_small": Object {
                "height": "100",
                "mp4": "https://media0.giphy.com/media/3bztNGb4XTnwxOgotG/100.mp4",
                "mp4_size": "45632",
                "size": "224834",
                "url": "https://media0.giphy.com/media/3bztNGb4XTnwxOgotG/100.gif",
                "webp": "https://media0.giphy.com/media/3bztNGb4XTnwxOgotG/100.webp",
                "webp_size": "64128",
                "width": "178",
              },
              "fixed_height_small_still": Object {
                "height": "100",
                "size": "15905",
                "url": "https://media0.giphy.com/media/3bztNGb4XTnwxOgotG/100_s.gif",
                "width": "178",
              },
              "fixed_height_still": Object {
                "height": "200",
                "size": "53389",
                "url": "https://media0.giphy.com/media/3bztNGb4XTnwxOgotG/200_s.gif",
                "width": "356",
              },
              "fixed_width": Object {
                "height": "113",
                "mp4": "https://media0.giphy.com/media/3bztNGb4XTnwxOgotG/200w.mp4",
                "mp4_size": "31518",
                "size": "274402",
                "url": "https://media0.giphy.com/media/3bztNGb4XTnwxOgotG/200w.gif",
                "webp": "https://media0.giphy.com/media/3bztNGb4XTnwxOgotG/200w.webp",
                "webp_size": "76074",
                "width": "200",
              },
              "fixed_width_downsampled": Object {
                "height": "113",
                "size": "109573",
                "url": "https://media0.giphy.com/media/3bztNGb4XTnwxOgotG/200w_d.gif",
                "webp": "https://media0.giphy.com/media/3bztNGb4XTnwxOgotG/200w_d.webp",
                "webp_size": "27014",
                "width": "200",
              },
              "fixed_width_small": Object {
                "height": "57",
                "mp4": "https://media0.giphy.com/media/3bztNGb4XTnwxOgotG/100w.mp4",
                "mp4_size": "12061",
                "size": "72074",
                "url": "https://media0.giphy.com/media/3bztNGb4XTnwxOgotG/100w.gif",
                "webp": "https://media0.giphy.com/media/3bztNGb4XTnwxOgotG/100w.webp",
                "webp_size": "27242",
                "width": "100",
              },
              "fixed_width_small_still": Object {
                "height": "57",
                "size": "6211",
                "url": "https://media0.giphy.com/media/3bztNGb4XTnwxOgotG/100w_s.gif",
                "width": "100",
              },
              "fixed_width_still": Object {
                "height": "113",
                "size": "19281",
                "url": "https://media0.giphy.com/media/3bztNGb4XTnwxOgotG/200w_s.gif",
                "width": "200",
              },
              "looping": Object {
                "mp4": "https://media0.giphy.com/media/3bztNGb4XTnwxOgotG/giphy-loop.mp4",
                "mp4_size": "1696420",
              },
              "original": Object {
                "frames": "17",
                "hash": "be8969113c5bb186d6e22fadda72ac9e",
                "height": "270",
                "mp4": "https://media0.giphy.com/media/3bztNGb4XTnwxOgotG/giphy.mp4",
                "mp4_size": "127017",
                "size": "1556485",
                "url": "https://media0.giphy.com/media/3bztNGb4XTnwxOgotG/giphy.gif",
                "webp": "https://media0.giphy.com/media/3bztNGb4XTnwxOgotG/giphy.webp",
                "webp_size": "304532",
                "width": "480",
              },
              "original_mp4": Object {
                "height": "270",
                "mp4": "https://media0.giphy.com/media/3bztNGb4XTnwxOgotG/giphy.mp4",
                "mp4_size": "127017",
                "width": "480",
              },
              "original_still": Object {
                "height": "270",
                "size": "95193",
                "url": "https://media0.giphy.com/media/3bztNGb4XTnwxOgotG/giphy_s.gif",
                "width": "480",
              },
              "preview": Object {
                "height": "136",
                "mp4": "https://media0.giphy.com/media/3bztNGb4XTnwxOgotG/giphy-preview.mp4",
                "mp4_size": "24070",
                "width": "241",
              },
              "preview_gif": Object {
                "height": "68",
                "size": "48823",
                "url": "https://media0.giphy.com/media/3bztNGb4XTnwxOgotG/giphy-preview.gif",
                "width": "121",
              },
              "preview_webp": Object {
                "height": "112",
                "size": "47494",
                "url": "https://media0.giphy.com/media/3bztNGb4XTnwxOgotG/giphy-preview.webp",
                "width": "199",
              },
            },
            "import_datetime": "2018-04-06 17:39:41",
            "isFavorite": true,
            "is_sticker": 0,
            "rating": "g",
            "saving": true,
            "slug": "cbs-lol-mom-3bztNGb4XTnwxOgotG",
            "source": "",
            "source_post_url": "",
            "source_tld": "",
            "title": "anna faris lol GIF by CBS",
            "trending_datetime": "2018-04-07 06:15:01",
            "type": "gif",
            "url": "https://giphy.com/gifs/cbs-lol-mom-3bztNGb4XTnwxOgotG",
            "user": Object {
              "avatar_url": "https://media.giphy.com/avatars/cbs/Ow00eLqoajbR.jpg",
              "banner_url": "https://media.giphy.com/headers/cbs/9tl9GSQXcvmD.jpg",
              "display_name": "CBS",
              "is_verified": true,
              "profile_url": "https://giphy.com/cbs/",
              "twitter": "@CBS",
              "username": "cbs",
            },
            "username": "cbs",
          },
          Object {
            "id": "ulKtDV83bDAz5kcXAW",
            "images": Object {
              "fixed_width": Object {
                "height": "356",
                "mp4": "https://media0.giphy.com/media/ulKtDV83bDAz5kcXAW/200w.mp4",
                "mp4_size": "263879",
                "size": "1923175",
                "url": "https://media0.giphy.com/media/ulKtDV83bDAz5kcXAW/200w.gif",
                "webp": "https://media0.giphy.com/media/ulKtDV83bDAz5kcXAW/200w.webp",
                "webp_size": "602334",
                "width": "200",
              },
              "original": Object {
                "frames": "45",
                "hash": "ef5ca4f048e3fb0840805d85ade6e04e",
                "height": "480",
                "mp4": "https://media0.giphy.com/media/ulKtDV83bDAz5kcXAW/giphy.mp4",
                "mp4_size": "411014",
                "size": "3238732",
                "url": "https://media0.giphy.com/media/ulKtDV83bDAz5kcXAW/giphy.gif",
                "webp": "https://media0.giphy.com/media/ulKtDV83bDAz5kcXAW/giphy.webp",
                "webp_size": "897330",
                "width": "270",
              },
            },
            "isFavorite": true,
            "saving": true,
            "title": "dog puppy GIF by Laurentian University",
            "type": "gif",
          },
        ],
        "error": "",
        "fetching": false,
        "pagination": Object {
          "count": 25,
          "offset": 1,
          "pages": 1,
          "total_count": 2,
        },
        "query": "",
        "selected2Add": Object {},
        "selectedID2remove": "",
      }
    }
  />
</MemoryRouter>
`;
